### PR TITLE
chore(azure regions): support non default azure region

### PR DESCRIPTION
**Pull request by n4ch04**
### Context

Prowler currently does not support non-default Azure regions

### Description

Support the following regions for the `azure` provider
```
 - AZURE_CHINA_CLOUD
 - AZURE_GERMAN_CLOUD
 - AZURE_US_GOV_CLOUD
```

### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by n4ch04** [comment]
Missing tests and documentation

**Comment by n4ch04** [comment]
Documentation and tests from new function added. It works for different cloud Azure accounts. Missing azure provider tests will be added in another PR

File: docs/tutorials/azure/use-non-default-region.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# Use non default Azure regions
+
+Microsoft provides clouds for compliance with regional laws, which are available for your use.
+By default, Prowler uses `AzureCloud` cloud which is the comercial one. (you can list all the available with `az cloud list --output table`).
+
+At the time of writing this doc the available clouds from different regions are:
```
Comment (jfagoagas) [inline docs/tutorials/azure/use-non-default-region.md]:
```suggestion
At the time of writing this documentation the available Azure Clouds from different regions are the following:
```

File: docs/tutorials/azure/use-non-default-region.md
Comment (jfagoagas) [inline docs/tutorials/azure/use-non-default-region.md:1]: I see this file but I think you have to include it in the index at the `mkdocs.yaml` file.

File: prowler/providers/azure/azure_provider.py [hunk 2]
```diff
@@ -32,6 +35,9 @@ def __init__(
             subscription_ids,
         )
 
+    def __set_region_config__(self, region):
```
Comment (jfagoagas) [inline prowler/providers/azure/azure_provider.py]: Why this function is called `__set_region_config__` but then is calling `get_regions_config`, maybe you can call directly `get_regions_config`

File: prowler/providers/azure/lib/arguments/arguments.py [hunk 3]
```diff
@@ -40,3 +43,27 @@ def init_parser(self):
...
+    azure_regions_subparser.add_argument(
+        "--azure-region",
+        nargs="?",
+        default="AzureCloud",
+        type=validate_azure_region,
+        help="Azure region from `az cloud list --output table`, by default AzureCloud",
+    )
+
+
+def validate_azure_region(region):
```
Comment (jfagoagas) [inline prowler/providers/azure/lib/arguments/arguments.py:57]: Can you include tests for this function?

**Review by jfagoagas — CHANGES_REQUESTED** [review]
Great addition @n4ch04 🚀 

Please check my comments then we can merge, thanks! 👏

File: docs/tutorials/azure/use-non-default-region.md (hunk 1 above)
Comment (n4ch04) [reply docs/tutorials/azure/use-non-default-region.md]: Done

File: docs/tutorials/azure/use-non-default-region.md
Comment (n4ch04) [reply docs/tutorials/azure/use-non-default-region.md:1]: Done

File: prowler/providers/azure/azure_provider.py (hunk 2 above)
Comment (n4ch04) [reply prowler/providers/azure/azure_provider.py]: Done

File: prowler/providers/azure/lib/arguments/arguments.py (hunk 3 above)
Comment (n4ch04) [reply prowler/providers/azure/lib/arguments/arguments.py:57]: Done

File: mkdocs.yml [hunk 4]
```diff
@@ -56,6 +56,7 @@ nav:
           - Boto3 Configuration: tutorials/aws/boto3-configuration.md
       - Azure:
           - Authentication: tutorials/azure/authentication.md
+          - Non defaul clouds: tutorials/azure/use-non-default-cloud.md
```
Comment (jfagoagas) [inline mkdocs.yml]:
```suggestion
          - Non default clouds: tutorials/azure/use-non-default-cloud.md
```

**Review by jfagoagas — DISMISSED** [review]
🚀 🚀 🚀 🚀

**Review by jfagoagas — DISMISSED** [review]

**Review by jfagoagas — APPROVED** [review]

**Pull request merged**